src/protocol-handlers.js

```diff
@@ -1,31 +1,215 @@
-const { app, shell } = require('electron')
+const { app, shell, ipcMain } = require('electron')
```

Comment:
Many of  these changes seem unrelated to the general setting up of protocol-handlers. Is there a better place for a majority of the logic required to set up the IPFS protocol?

Maybe move `src/protocol-handlers.js` to `src/protocol-handlers/index.js` and then the logic for the ipfs protocol handling inside of `src/protocol-handlers/ipfs.js` ?